### PR TITLE
Issue-5191 Copy android assets to correct folder when creating AAB

### LINKED ISSUE #5191: Android App Bundles copy assets to the wrong directory in the aab file
**Describe the bug (REQUIRED)**
From user: "Create extension with res\android\assets\file_in_assets.txt. Bundle it as apk and you can find this file in assets directory of apk. But if you bundle it as aab you can find this file in base\assets\assets directory of aab."

**To Reproduce (REQUIRED)**
* Create extension with res\android\assets\file_in_assets.txt
* Bundle as APK and you can find this file in assets directory of apk
* Bundle it as AAB you can find this file in base\assets\assets directory

**Expected behavior (REQUIRED)**
There should not be duplicate assets/assets folder

**Defold version (REQUIRED):**
  - 1.2.172

### PR DESCRIPTION
**Pull request by britzl**
Fixes #5191

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java [hunk 1]
```diff
@@ -124,6 +124,15 @@ private static Result exec(String... args) throws IOException {
         return exec(Arrays.asList(args));
     }
 
+    private static CompileExceptionError createCompileExceptionError(String message) {
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java:127]: I decided to clean up how CompileExceptionErrors were created as I didn't get proper stack traces before

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java [hunk 2]
```diff
@@ -487,9 +497,28 @@ private static File createAppBundleBaseZip(Project project, File outDir, File ap
...
+                File file = null;
+                if (filename.startsWith(assets)) {
+                    file = new File(assetsDir, filename.substring(assets.length()));
+                }
+                else {
+                    file = new File(rootDir, filename);
+                }
+                log("Copying resource to " + file);
+                ExtenderUtil.writeResourceToFile(resource, file);
+            }
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/AndroidBundler.java:520]: The actual fix. Files starting with assets/ should be copied to the assets/ dir and files starting with something else should be copied to the root/ dir.

**Review by JCash — DISMISSED** [review]
Looks good!

File: com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/bundle/test/BundlerTest.java [hunk 3]
```diff
@@ -648,23 +656,37 @@ public void testBundleResourcesDirs() throws IOException, ConfigurationException
         sub_platform2.mkdir();
         createDefaultFiles(contentRoot);
         createFile(contentRoot, "m.txt", "dummy");
-        createFile(sub_platform1.getAbsolutePath(), "s1-1.txt", "dummy");
-        createFile(sub_platform1.getAbsolutePath(), "s1-2.txt", "dummy");
-        createFile(sub_platform2.getAbsolutePath(), "s2-1.txt", "dummy");
-        createFile(sub_platform2.getAbsolutePath(), "s2-2.txt", "dummy");
+        createFile(sub_platform1.getAbsolutePath(), "s1-root1.txt", "dummy");
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob.test/src/com/dynamo/bob/bundle/test/BundlerTest.java:659]: Renamed to help with debugging as the name itself now conveys where it is located

**Review by JCash — APPROVED** [review]

**Pull request merged**